### PR TITLE
PF-994: Define CLI enum for workspace user roles.

### PR DESCRIPTION
**Pull request by marikomedlock**
Defined a CLI enum for workspace user roles: READER, WRITER, OWNER. It is different from the WSM enum for workspace user roles, which also includes the APPLICATION role.

We don't want to expose the APPLICATION role to CLI users at this time. There are no immediate use cases for it, and possible future use cases (e.g. Leo) may not use the CLI to setup the application user anyway. So for now, I've hidden this role from CLI users to avoid confusion. Previously trying to define an application user resulted in an error.

Also fixed an NPE when listing referenced resources in text format.

### CODE REVIEW

**Review by jmczerk — APPROVED** [review]

File: src/main/java/bio/terra/cli/serialization/userfacing/UFWorkspaceUser.java [hunk 1]
```diff
@@ -36,21 +36,21 @@ private UFWorkspaceUser(Builder builder) {
   /** Print out this object in text format. */
   public void print() {
     PrintStream OUT = UserIO.getOut();
-    List<String> rolesStr = roles.stream().map(IamRole::toString).collect(Collectors.toList());
+    List<String> rolesStr = roles.stream().map(Role::toString).collect(Collectors.toList());
     OUT.println(email + ": " + String.join(",", rolesStr));
```
Comment (jaycarlton) [inline src/main/java/bio/terra/cli/serialization/userfacing/UFWorkspaceUser.java:40]: nit: you could do the join in your collect statement. collectors.joining or some such.

File: src/test/java/harness/utils/CleanupTestUserWorkspaces.java [hunk 2]
```diff
@@ -61,7 +61,8 @@ private static void deleteWorkspaces(TestUsers testUser, boolean isDryRun) throw
                 .filter(user -> user.email.equalsIgnoreCase(testUser.email))
                 .findAny();
         // skip deleting if the test user is not an owner
-        if (workspaceUser.isEmpty() || !workspaceUser.get().roles.contains(IamRole.OWNER)) {
+        if (workspaceUser.isEmpty()
+            || !workspaceUser.get().roles.contains(WorkspaceUser.Role.OWNER)) {
```
Comment (jaycarlton) [inline src/test/java/harness/utils/CleanupTestUserWorkspaces.java:65]: nit. Instead of checking empty first you can do map/contains/orElse(false).

**Review by jaycarlton — APPROVED** [review]

File: src/main/java/bio/terra/cli/businessobject/WorkspaceUser.java [hunk 3]
```diff
@@ -22,9 +22,39 @@
...
-  private WorkspaceUser(String email, List<IamRole> roles) {
+  /**
+   * Enum for the workspace user roles supported by the CLI. Each enum value maps to a single WSM
+   * client library ({@link bio.terra.workspace.model.IamRole}) enum value.
+   *
+   * <p>The CLI defines its own enum instead of using the WSM one so that we can restrict the roles
+   * supported (e.g. no applications). It also gives the CLI control over what the enum names are,
+   * which are exposed to users as command options.
+   */
+  public enum Role {
```
[the remark this answers is not in the document]
Comment (marikomedlock) [reply src/main/java/bio/terra/cli/businessobject/WorkspaceUser.java:35]: The `UF` prefix stands for "user-facing" and is used for serialization POJO classes that are exposed to the CLI users, either as inputs (e.g. GCS bucket lifecycle rules json format) or outputs (e.g. workspace object). Similarly, the `PD` prefix stands for "persisted" and is used for serialization POJO classes that are persisted by the CLI in its context directory. I haven't used prefixes for internal business object classes yet.

File: src/main/java/bio/terra/cli/serialization/userfacing/UFWorkspaceUser.java (hunk 1 above)
Comment (marikomedlock) [reply src/main/java/bio/terra/cli/serialization/userfacing/UFWorkspaceUser.java:40]: done.

File: src/test/java/harness/utils/CleanupTestUserWorkspaces.java (hunk 2 above)
Comment (marikomedlock) [reply src/test/java/harness/utils/CleanupTestUserWorkspaces.java:65]: I find having it written out slightly easier to follow. just personal preference.

**Pull request merged**